src/components/routerView.ts

```diff
@@ -9,22 +10,28 @@ export default defineComponent({
   setup() {
     const router = inject(routerInjectionKey)
     const depth = inject(depthInjectionKey, 0)
+    const routerView = resolveComponent('RouterView')
 
     if (!router) {
       throw new Error('Router not installed')
     }
 
     provide(depthInjectionKey, depth + 1)
 
-    const routerView = resolveComponent('RouterView')
-
     const component = computed(() => {
-      const route = router.route.matches[depth]
+      const routeComponent = router.route.matches[depth].component
+
+      if (routeComponent) {
+        if (typeof routeComponent === 'function') {
+          return defineAsyncComponent(routeComponent as AsyncComponentLoader)
+        }
+
+        return routeComponent as RouteComponent
```

Comment:
in my testing this line doesn't need `as RouteComponent`

Reply:
Nice, idk what I changed to make that work now but happy to remove an `as`.